<?xml version="1.0"?>

<launch>

    <!-- Options -->
    <arg name="model_name"          default="FastSAM-x.pt"/>
    <arg name="use_cuda"            default="true"/>
    <arg name="conf"                default="0.4"/>
    <arg name="iou"                 default="0.9"/>
    <arg name="prompt_type"         default="bbox"/>
    <arg name="publish_ann"         default="false"/>
    <arg name="frame_id"            default="map"/>
    <arg name="verbose"             default="false"/>

    <!-- Consumed topics -->
    <arg name="rgb_topic"           default="image"/>
    <arg name="pointcloud_topic"    default="pointcloud"/>

    <!-- Produced topics -->
    <arg name="sidewalk_mask_topic"         default="sidewalk_mask"/>
    <arg name="sidewalk_pointcloud_topic"   default="sidewalk_pointcloud"/>
    <arg name="sidewalk_ann_topic"          default="sidewalk_ann"/>

    <!-- Nodes -->
    <node name="sidewalk_segmentation" pkg="svea_vision" type="sidewalk_segmentation.py" output="screen">
        <!-- Options -->
        <param name="model_name"                    value="$(arg model_name)" />
        <param name="use_cuda"                      value="$(arg use_cuda)" />
        <param name="conf"                          value="$(arg conf)" />
        <param name="iou"                           value="$(arg iou)" />
        <param name="prompt_type"                   value="$(arg prompt_type)" />
        <param name="publish_ann"                   value="$(arg publish_ann)" />
        <param name="frame_id"                      value="$(arg frame_id)" />
        <param name="verbose"                       value="$(arg verbose)" />
        <!-- Topics -->
        <param name="rgb_topic"                     value="$(arg rgb_topic)" />
        <param name="pointcloud_topic"              value="$(arg pointcloud_topic)" />
        <param name="sidewalk_mask_topic"           value="$(arg sidewalk_mask_topic)" />
        <param name="sidewalk_pointcloud_topic"     value="$(arg sidewalk_pointcloud_topic)" />
        <param name="sidewalk_ann_topic"            value="$(arg sidewalk_ann_topic)" />
    </node>

</launch>